src/Abstractions/NexusMods.Abstractions.Loadouts/Loadout.cs

```diff
@@ -48,160 +49,58 @@ public static class Loadout
-    {
-        // All db revisions that contain the loadout id, select the loadout
-        return conn.Revisions
-            .Where(db => db.Datoms(db.BasisTxId).Any(datom => datom.E == id.Value))
-            .StartWith(conn.Db)
-            .Select(db => db.Get<Model>(id.Value));
-    }
-
+    public static readonly BackReferenceAttribute<Mod> Mods = new(Mod.Loadout);
+    
     /// <summary>
-    /// Retracts a loadout, performing an effective deletion.
+    /// All the files that are part of this loadout point to this entity via File.Loadout
     /// </summary>
-    /// <param name="conn">The connection to use.</param>
-    /// <param name="loadoutId">The ID of the loadout which needs to be retracted.</param>
-    public static async Task Delete(this IConnection conn, LoadoutId loadoutId)
-    {
-        using var db = conn.Db;
-        var loadout = db.Get<Model>(loadoutId.Value);
-
-        using var tx = conn.BeginTransaction();
-
-        // Retract the loadout itself by changing its kind to `Deleted`
-        // This marks the entity for Garbage Collection on next GC run.
-        LoadoutKind.Add(tx, loadout.Id, Abstractions.Loadouts.LoadoutKind.Deleted, false);
-        loadout.Revise(tx);
-        await tx.Commit();
-    }
+    public static readonly BackReferenceAttribute<File> Files = new(File.Loadout);
     
-    public class Model(ITransaction tx) : Entity(tx)
+    public partial struct ReadOnly
     {
-        /// <summary>
-        /// The unique identifier for this loadout, casted to a <see cref="LoadoutId"/>.
-        /// </summary>
-        public LoadoutId LoadoutId => LoadoutId.From(Id);
-
-        /// <summary>
-        /// Gets the loadout id/txid pair for this revision of the loadout.
-        /// </summary>
-        public LoadoutWithTxId GetLoadoutWithTxId()
-        {
-            return new LoadoutWithTxId(LoadoutId, GetRevisionTxId());
-        }
-        
-        /// <summary>
-        /// The most recent transaction Id that modified this loadout model
-        /// </summary>
-        public TxId GetRevisionTxId() => Db.Datoms(Id).Max(d => d.T);
-
-        public string Name
-        {
-            get => Loadout.Name.Get(this);
-            set => Loadout.Name.Add(this, value);
-        }
-        
-        /// <summary>
-        /// Get the installation id for this loadout.
-        /// </summary>
-        public EntityId InstallationId
-        {
-            get => Loadout.Installation.Get(this);
-            set => Loadout.Installation.Add(this, value);
-        }
-
-        /// <summary>
-        /// The revision number for this loadout, increments by one for each change.
-        /// </summary>
-        public ulong Revision
+        public override bool Equals(object? obj)
         {
-            get => Loadout.Revision.Get(this, 0);
-            set => Loadout.Revision.Add(this, value);
+            if (obj is not ReadOnly other)
+                return false;
+            return Id.Equals(other.Id);
```

Comment:
Why do we override this method? The definition of equality isn't really made clear to the consumer, since all they see is `Equals(object? obj)`.

Since `Loadout.ReadOnly` is a value type, it will also get boxed when using this method.

I think a better solution would be to source generate an implementation of `IEqualityComparer<Loadout.ReadOnly>` like this:

```csharp
public class IdEqualityComparer : IEqualityComparer<Loadout.ReadOnly>
{
  public static readonly IEqualityComparer<Loadout.ReadOnly> Instance = new IdEqualityComparer();

  //...
}
```

With this approach, you can explicitly use this comparer in collections or other operations.